posts/_markdown/vqa_in_qutip.md

```diff
@@ -0,0 +1,106 @@
+
+## Module Overview
+
+Most quantum algorithms deal with quantum circuits, which are made up of quantum gates. In `QuTiP`, there exist classes --- `QubitCircuit` and `Gate` --- which simulate the dynamics of quantum circuits.
+
+For VQAs, we now want these gates to be parameterized. In addition to the gate parameters, we want initialization options, which are fixed throughout the optimization process, to be easily defined in one object. For these purposes, we define two new classes, `VQA_Block` and `VQA`, which are abstractions on `Gate` and `QubitCircuit`, respectively.
+
+### 1. The VQA_Block class
+
+The `VQA_Block` class encapsulates one component of our circuit and can store the action of one or more quantum gates. It can hold parameters, or keep note of how many free parameters it needs as input to be translated into a gate on the quantum circuit.
+
+In quantum computing, all quantum gates can be described by unitary operators. Our `VQA_Block` class has a method that will generate a unitary operator for the quantum circuit; but crucially, the value of this operator can change based on the parameters the `VQA_Block` receives.
+
+The `VQA_Block` holds information about the operator it can generate, the parameters it needs, and methods to compute derivates of its operator. Using this information, the algorithm can compute gradients of its cost function. The optimization process is thus a loop in which `VQA_Block`s are evaluated to inform the minimization of the cost of a circuit.
+
+### 2. The VQA class
+
+The `VQA` class holds information about the problem being optimized. It has methods to run optimization procedures with various options, such as the number of qubits required, the number of layers --- or repetitions --- of the circuit, and the optimization method to use. Here, we allow access to all of the function minimization methods available within [the `scipy` library](https://scipy.org/), as well as allowing user-defined optimization methods.
+
+One part of the optimization process is specifying a "cost function" which the optimizer can use to assess the performance of a particular parameter set. In the case of a real quantum computer, the only output we have comes from taking measurements of the system. In quantum computing, these are usually represented as "bitstrings", which encode the measurement outcome. For example, the measurement outcome of a two qubit system with one qubit measured in the |0> state, and another in the |1> state can be represented by the bitstring |01>.
+
+If the cost method is set to `BITSTRING`, then the user needs to provide a function that takes in this string of 1's and 0's, and returns the associated cost.
+
+Because we're performing a quantum simulation, we don't just have to deal with measurements --- we can "peak under the hood" of the simulation to get more information, including the actual state of the system before measurement. With this, we allow two more methods for specifying a cost:
```

Comment:
```suggestion
Because we're performing a quantum simulation, we don't just have to deal with measurements --- we can "peek under the hood" of the simulation to get more information, including the actual state of the system before measurement. With this, we allow two more methods for specifying a cost:
```